android/src/com/frostwire/android/gui/activities/internal/NavigationMenu.java

```diff
@@ -0,0 +1,234 @@
+import android.support.v7.widget.Toolbar;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import com.frostwire.android.BuildConfig;
+import com.frostwire.android.R;
+import com.frostwire.android.core.Constants;
+import com.frostwire.android.gui.SoftwareUpdater;
+import com.frostwire.android.gui.activities.BuyActivity;
+import com.frostwire.android.gui.activities.MainActivity;
+import com.frostwire.android.gui.services.Engine;
+import com.frostwire.android.gui.util.UIUtils;
+import com.frostwire.android.gui.views.AdMenuItemView;
+import com.frostwire.android.offers.Offers;
+
+/**
+ * @author aldenml
+ * @author gubatron
+ * @author marcelinkaaa
+ *
+ * Created on 02/23/2017
+ */
+
+public class NavigationMenu {
+    private final MainController controller;
+    private final NavigationView navView;
+    private final DrawerLayout drawerLayout;
+    private final ActionBarDrawerToggle drawerToggle;
+    private final AdMenuItemView menuRemoveAdsItem;
+    private int checkedNavViewMenuItemId = -1;
+
+    public NavigationMenu(MainController controller, DrawerLayout drawerLayout, Toolbar toolbar) {
+        this.controller = controller;
+        this.drawerLayout = drawerLayout;
+        MainActivity mainActivity = controller.getActivity();
+        drawerToggle = new MenuDrawerToggle(controller, drawerLayout, toolbar);
+        this.drawerLayout.addDrawerListener(drawerToggle);
+        navView = initNavigationView(mainActivity);
+        menuRemoveAdsItem = initAdMenuItemListener(mainActivity);
+        refreshMenuRemoveAdsItem();
+    }
+
+    public boolean isOpen() {
+        return drawerLayout.isDrawerOpen(navView);
+    }
+
+    public void show() {
+        drawerLayout.openDrawer(navView);
+    }
+
+    public void hide() {
+        drawerLayout.closeDrawer(navView);
+    }
+
+    public void onConfigurationChanged(Configuration newConfig) {
+        drawerToggle.onConfigurationChanged(newConfig);
+    }
+
+    public void syncState() {
+        drawerToggle.syncState();
+    }
+
+    public void updateCheckedItem(int menuItemId) {
+        navView.setCheckedItem(menuItemId);
+    }
+
+    private NavigationView initNavigationView(final Activity activity) {
+        NavigationView resultNavView = navView;
+        if (navView == null) {
+            resultNavView = (NavigationView) activity.findViewById(R.id.activity_main_nav_view);
+            resultNavView.setNavigationItemSelectedListener(
+                    new NavigationView.OnNavigationItemSelectedListener() {
+                        @Override
+                        public boolean onNavigationItemSelected(MenuItem menuItem) {
+                            onMenuItemSelected(menuItem);
+                            return true;
+                        }
+                    });
+            View navViewHeader = resultNavView.getHeaderView(0);
```

Comment:
works for me on android 4.4, it'd be awesome to keep this and start cleaning up any redundant icons (if what this does is to programatically paint icons)
![screen shot 2017-02-25 at 11 06 02 am](https://cloud.githubusercontent.com/assets/163977/23333557/8cf70598-fb4a-11e6-8165-2abbbfc72c87.png)